### DISCUSSION TITLE
A syntax for querying of metrics1/metrics2 calculation ratio where metrics1 and metrics2 are diffrerent metrics sharing the same capturing group in regex.

### DISCUSSION THREAD

**Discussion by Eljah** [prometheus/prometheus#10351] [Q&A] [answered]
I have a couple of metrics like:

camel_proxy_feature1_sum
camel_proxy_feature1_count
camel_proxy_feature2_sum
camel_proxy_feature2_count

I'd like to plot the calculation of camel_proxy.*sum/camel_proxy.*count and for now can't figure out how to query a vector for that correctly

If I try 
```
label_replace({__name__=~"camel_proxy.*sum"},"name_label","$1","__name__", "(.+)")/label_replace({__name__=~"camel_proxy.*count"},"name_label","$1","__name__", "(.+)")
```
it returns empty value and it is clear why, it is impossible to sync one group with another. So in fact I need somehow to request for values like feature1, feature2 and them form the vector for camel_proxy_feature1_sum/camel_proxy_feature1_count, camel_proxy_feature2_sum/camel_proxy_feature2_count and so on.

But I'm pretty sure it is impossible for now.

**Comment by beorn7** [comment, marked as answer]
Does it work if you change the regexp (last parameter of `label_replace`) to `(.+)_(sum|count)`?

**Comment by Eljah** [comment]
@beorn7 yes, that works (als in more details here: https://stackoverflow.com/questions/71158998/how-to-plot-a-complex-combination-of-several-prometheus-metrics-obeying-the-same/71228271#71228271)

but could you please explain why it works?

**Comment by beorn7** [comment]
With your original query, you put the whole metric name into the new `name_label`, including the `_sum`/`_count` suffix. Therefore, the `name_label` values don't match between enumerator and denominator in the ratio. With my suggested modification, you exclude the suffix, and now the `name_label` values match.

`label_replace` is hard to master. One day, we'll have a really nice tutorial on https://prometheus.io . Until then, you have to search the web. There are many explanations out there.
